backend/src/middleware/validUrlMiddleware.js

```diff
@@ -0,0 +1,18 @@
+const validURL = str => {
+  const isValid = str.match(/^(?:https?:\/\/)(?:[^@\n])?(?:www\.)?([^:/\n?]+)/g)
```

Comment:
Would be great to see this unit tested. No need for the database to be involved.

Reply:
E.g. I wonder if https is required or not.

Reply:
and what about a missing domain? Like: `www.google`

Reply:
I am not an expert on regex, but I believe I tested this and the `s` is optional, which is why it has a `?` on the right of it?

Reply:
not sure what you mean by missing domain... as in no `.com`, etc?
I think this is the regex you came up with and I copied from `my-social-media.vue`... could be that we could improve on it...

Reply:
I'm sorry, I meant **top-level**-domain.

Reply:
Up to you! If you want to get the PR merged, maybe create `good-first-issue` for refactoring?

Reply:
can be, that sounds like a good compromise that would potentially allow more conversation on the subject